### PR TITLE
feat(server): allow to add a description field to search results

### PR DESCRIPTION
**Pull request by dlehmhus**
Allow defining a path to a description field in the model, similar to the already existing title path. If the path is set, the value of that field will be included in a content search response as description.

<hr /><p><time>(5/16/2019, 9:48:52 AM)</date> Pre-released as <code>@cotype/core@1.7.0-beta.8bfe65a</code></p>

### CODE REVIEW

File: src/content/rest/__tests__/restApi.test.ts [hunk 1]
```diff
@@ -366,6 +366,43 @@ describe("rest api", () => {
...
+          })).items[0]
+        ).toMatchObject({ description });
+
+        expect(
+          (await search(description, {
+            published: false,
+            linkableOnly: false,
+            includeModels: ["articleNews"]
+          })).items[0]
+        ).not.toMatchObject({ description });
```
Comment (Xiphe) [inline src/content/rest/__tests__/restApi.test.ts]: I would extract this second assertion in a separate test. "it should not contain description in result when no path is provided in model"
And also change the assertion to 
```ts
expect((await search/* ... */).items[0].description).toBe(undefined);
```

File: src/content/rest/__tests__/restApi.test.ts [hunk 2]
```diff
@@ -366,6 +366,43 @@ describe("rest api", () => {
...
+          slug: "foo-bar",
+          title: description
+        });
+
+        expect(
+          (await search(description, {
+            published: false,
+            linkableOnly: false
+          })).total
+        ).toBe(2);
```
Comment (Xiphe) [inline src/content/rest/__tests__/restApi.test.ts]: This assertion seems unrelated to the test case

File: src/content/formatQuillDelta.ts [hunk 3]
```diff
@@ -8,10 +8,12 @@ import { QuillDelta } from "../../typings";
...
-  return data.ops.reduce((text, op) => {
-    if (typeof op.insert !== "string") return text + " ";
-    return text + op.insert;
-  }, "");
+  return data.ops
+    .reduce((text, op) => {
+      if (typeof op.insert !== "string") return text + " ";
+      return text + op.insert.replace(/(\r\n|\n|\r)/gm, " ");
+    }, "")
+    .trim();
```
Comment (Xiphe) [inline src/content/formatQuillDelta.ts]: As we discussed in person, let's not do this. If this makes trouble in the client, it should remove the line-breaks itself.

File: src/persistence/ContentPersistence.ts [hunk 4]
```diff
@@ -11,6 +11,20 @@ import getRefUrl from "../content/getRefUrl";
...
 import { getDeepJoins } from "../content/rest/filterRefData";
+import { ContentFormat } from "../../typings";
+
+function findValueByPath(path: string | undefined, data: Cotype.Data) {
+  if (!path) return;
+
+  const titlePath = path.split(".");
+
+  const title = (titlePath.reduce(
+    (obj, key) => (obj && obj[key] !== "undefined" ? obj[key] : undefined),
```
Comment (Xiphe) [inline src/persistence/ContentPersistence.ts]: I know this was just moved. Is the `"undefined"` string intended? From just looking at this code, i'd assume it should be `undefined`.

Comment (dlehmhus) [reply src/persistence/ContentPersistence.ts]: It seem as if a `typeof` is missing, otherwise the `"undefinded"` string would make no sense. Furthermore wouldn't just `obj ? obj[key] : undefined` do the trick?

Comment (Xiphe) [reply src/persistence/ContentPersistence.ts]: seems this way. But again maybe the API is not intuitive :)

**Comment by fgnass** [comment]
Sorry for chiming in on an already merged PR: What about returning an excerpt from `content_search.text` instead? This would allow clients to highlight the search term in a google-like manner.
